cadence/client.py

```diff
@@ -229,6 +231,74 @@ async def start_workflow(
         except Exception:
             raise
 
+    async def signal_with_start_workflow(
+        self,
+        workflow: Union[str, Callable],
```

Comment:
maybe we require workflow to be `WorkflowDefinition`?
```suggestion
        workflow: Union[str, WorkflowDefinition],
```